gb2seq/features.py

```diff
@@ -239,21 +246,27 @@ def _initializeFromGenBankRecord(
             elif type_ == "repeat_region":
                 for n in itertools.count(1):
                     name = f"repeat region {n}"
-                    if name not in self:
+                    if name not in self._data:
                         break
                 value["type"] = feature.qualifiers["rpt_type"][0]
 
+            elif "product" in feature.qualifiers:
+                # This covers "CDS", "mat_peptide", "ncRNA".
+                name = feature.qualifiers["product"][0]
+                value["product"] = name
+
             else:
-                if "product" in feature.qualifiers:
-                    # This covers "CDS", "mat_peptide", "ncRNA".
-                    name = feature.qualifiers["product"][0]
-                    value["product"] = name
+                # Skip unwanted features that are not annotated as having a product
+                # (e.g., "source", "gap", "gene", "misc_feature".)
+                if type_ in alsoInclude:
```

Comment:
This could also be combined into an `elif`

Reply:
True. Done. Thanks.